src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2056,7 +2057,9 @@ def aks_upgrade(cmd,    # pylint: disable=unused-argument, too-many-return-state
     instance.service_principal_profile = None
     instance.aad_profile = None
 
-    return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, name, instance)
+    headers = get_aks_custom_headers(aks_custom_headers)
```

Comment:
any validation, dedup on input header values?

Reply:
the validation is in get_aks_custom_headers